// angular-countdown-date-time/__ivy_ngcc__/fesm2015/angular-countdown-date-time.js
import { Injectable, Component, Input, ViewEncapsulation, NgModule, defineInjectable } from '@angular/core';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as ɵngcc0 from '@angular/core';
class AngularCountdownDateTimeService {
    constructor() { }
}
AngularCountdownDateTimeService.ɵfac = function AngularCountdownDateTimeService_Factory(t) { return new (t || AngularCountdownDateTimeService)(); };
AngularCountdownDateTimeService.ɵprov = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjectable({ token: AngularCountdownDateTimeService, factory: AngularCountdownDateTimeService.ɵfac, providedIn: 'root' });
/** @nocollapse */
AngularCountdownDateTimeService.ctorParameters = () => [];
/** @nocollapse */ AngularCountdownDateTimeService.ngInjectableDef = defineInjectable({ factory: function AngularCountdownDateTimeService_Factory() { return new AngularCountdownDateTimeService(); }, token: AngularCountdownDateTimeService, providedIn: "root" });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(AngularCountdownDateTimeService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return []; }, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class AngularCountdownDateTimeComponent {
    constructor() {
        debugger;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        debugger;
        this.endDate = new Date(this.endDateBind);
        setInterval((/**
         * @return {?}
         */
        () => {
            /** @type {?} */
            let end = this.endDate.toDateString();
            /** @type {?} */
            let start = new Date().toDateString();
            this.diff = Math.floor((this.endDate.getTime() - new Date().getTime()) / 1000);
            this.dayTextBind = this.dayText;
            this.hourTextBind = this.hourText;
            this.minutesTextBind = this.minutesText;
            this.secondsTextBind = this.secondsText;
            this.days = this.getDays(this.diff);
            this.hours = this.getHours(this.diff);
            this.minutes = this.getMinutes(this.diff);
            this.seconds = this.getSeconds(this.diff);
        }), 1000);
    }
    /**
     * @param {?} t
     * @return {?}
     */
    getDays(t) {
        /** @type {?} */
        let days;
        days = Math.floor(t / 86400);
        return ('0' + days).slice(-2);
    }
    /**
     * @param {?} t
     * @return {?}
     */
    getHours(t) {
        /** @type {?} */
        let days;
        /** @type {?} */
        let hours;
        days = Math.floor(t / 86400);
        t -= days * 86400;
        hours = Math.floor(t / 3600) % 24;
        return ('0' + hours).slice(-2);
    }
    /**
     * @param {?} t
     * @return {?}
     */
    getMinutes(t) {
        /** @type {?} */
        let days;
        /** @type {?} */
        let hours;
        /** @type {?} */
        let minutes;
        days = Math.floor(t / 86400);
        t -= days * 86400;
        hours = Math.floor(t / 3600) % 24;
        t -= hours * 3600;
        minutes = Math.floor(t / 60) % 60;
        return ('0' + minutes).slice(-2);
    }
    /**
     * @param {?} t
     * @return {?}
     */
    getSeconds(t) {
        /** @type {?} */
        let days;
        /** @type {?} */
        let hours;
        /** @type {?} */
        let minutes;
        /** @type {?} */
        let seconds;
        days = Math.floor(t / 86400);
        t -= days * 86400;
        hours = Math.floor(t / 3600) % 24;
        t -= hours * 3600;
        minutes = Math.floor(t / 60) % 60;
        t -= minutes * 60;
        seconds = t % 60;
        return ('0' + seconds).slice(-2);
    }
}
AngularCountdownDateTimeComponent.ɵfac = function AngularCountdownDateTimeComponent_Factory(t) { return new (t || AngularCountdownDateTimeComponent)(); };
AngularCountdownDateTimeComponent.ɵcmp = /*@__PURE__*/ ɵngcc0.ɵɵdefineComponent({ type: AngularCountdownDateTimeComponent, selectors: [["lib-angular-countdown-dateTime"]], inputs: { endDateBind: "endDateBind", dayText: "dayText", hourText: "hourText", minutesText: "minutesText", secondsText: "secondsText" }, decls: 23, vars: 8, consts: [[1, "count-down"], [1, "count-down-row"]], template: function AngularCountdownDateTimeComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵelementStart(2, "div");
        ɵngcc0.ɵɵelementStart(3, "span");
        ɵngcc0.ɵɵtext(4);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "div");
        ɵngcc0.ɵɵelementStart(6, "span");
        ɵngcc0.ɵɵtext(7);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(8, "div");
        ɵngcc0.ɵɵelementStart(9, "span");
        ɵngcc0.ɵɵtext(10);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(11, "div");
        ɵngcc0.ɵɵelementStart(12, "span");
        ɵngcc0.ɵɵtext(13);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(14, "div", 1);
        ɵngcc0.ɵɵelementStart(15, "div");
        ɵngcc0.ɵɵtext(16);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(17, "div");
        ɵngcc0.ɵɵtext(18);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(19, "div");
        ɵngcc0.ɵɵtext(20);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(21, "div");
        ɵngcc0.ɵɵtext(22);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.days, " ");
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.hours, " ");
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.minutes, " ");
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.seconds, " ");
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.dayTextBind, " ");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.hourTextBind, " ");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.minutesTextBind, " ");
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.secondsTextBind, " ");
    } }, styles: [".count-down{background:#fff;border:0 solid #dadada;border-radius:4.48px;position:relative;width:100%;padding:20px 0 10px}.count-down .count-down-row{display:flex;flex-direction:row;justify-content:center;align-items:center;text-align:center}.count-down .count-down-row>div{flex-grow:1;padding-bottom:10px;width:25%}.count-down span{letter-spacing:0;font-size:1.5em;color:#616161;text-align:center;font-weight:500}"], encapsulation: 2 });
/** @nocollapse */
AngularCountdownDateTimeComponent.ctorParameters = () => [];
AngularCountdownDateTimeComponent.propDecorators = {
    endDateBind: [{ type: Input }],
    dayText: [{ type: Input }],
    hourText: [{ type: Input }],
    minutesText: [{ type: Input }],
    secondsText: [{ type: Input }]
};
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(AngularCountdownDateTimeComponent, [{
        type: Component,
        args: [{
                selector: 'lib-angular-countdown-dateTime',
                template: "<div class=\"count-down\">\r\n  <div class=\"count-down-row\">\r\n    <div>\r\n      <span>\r\n        {{days}}\r\n      </span>\r\n    </div>\r\n    <div>\r\n      <span>\r\n        {{hours}}\r\n      </span>\r\n    </div>\r\n    <div>\r\n      <span>\r\n        {{minutes}}\r\n      </span>\r\n    </div>\r\n    <div>\r\n      <span>\r\n        {{seconds}}\r\n      </span>\r\n    </div>\r\n  </div>\r\n  <div class=\"count-down-row\">\r\n    <div>\r\n      {{dayTextBind}}\r\n    </div>\r\n    <div>\r\n      {{hourTextBind}}\r\n    </div>\r\n    <div>\r\n      {{minutesTextBind}}\r\n    </div>\r\n    <div>\r\n      {{secondsTextBind}}\r\n    </div>\r\n  </div>\r\n</div>\r\n\r\n",
                encapsulation: ViewEncapsulation.None,
                styles: [".count-down{background:#fff;border:0 solid #dadada;border-radius:4.48px;position:relative;width:100%;padding:20px 0 10px}.count-down .count-down-row{display:flex;flex-direction:row;justify-content:center;align-items:center;text-align:center}.count-down .count-down-row>div{flex-grow:1;padding-bottom:10px;width:25%}.count-down span{letter-spacing:0;font-size:1.5em;color:#616161;text-align:center;font-weight:500}"]
            }]
    }], function () { return []; }, { endDateBind: [{
            type: Input
        }], dayText: [{
            type: Input
        }], hourText: [{
            type: Input
        }], minutesText: [{
            type: Input
        }], secondsText: [{
            type: Input
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class AngularCountdownDateTimeModule {
}
AngularCountdownDateTimeModule.ɵfac = function AngularCountdownDateTimeModule_Factory(t) { return new (t || AngularCountdownDateTimeModule)(); };
AngularCountdownDateTimeModule.ɵmod = /*@__PURE__*/ ɵngcc0.ɵɵdefineNgModule({ type: AngularCountdownDateTimeModule });
AngularCountdownDateTimeModule.ɵinj = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjector({ imports: [[]] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(AngularCountdownDateTimeModule, [{
        type: NgModule,
        args: [{
                declarations: [AngularCountdownDateTimeComponent],
                imports: [],
                exports: [AngularCountdownDateTimeComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AngularCountdownDateTimeModule, { declarations: [AngularCountdownDateTimeComponent], exports: [AngularCountdownDateTimeComponent] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { AngularCountdownDateTimeService, AngularCountdownDateTimeComponent, AngularCountdownDateTimeModule };
